neurolib/utils/collections.py

```diff
@@ -126,12 +128,17 @@ def unwrap_star_dotdict(dct, model, replaced_dict=False):
     E.g. params["*tau"] = 2.3 => params["mass1.tau"] = 2.3 and params["mass2.tau] = 2.3
     """
     # for each `k` that possibly contain stars get all key_u (unwrapped keys) from the star_dotdict
-    if replaced_dict:
-        return {
-            key_u: v for k, v in dct.items() for key_u in list(model.params[_sanitize_keys(k, replaced_dict)].keys())
-        }
-    else:
-        return {key_u: v for k, v in dct.items() for key_u in list(model.params[k].keys())}
+    return_dct = {}
+    for k, v in dct.items():
+        try:
+            for key_u in list(model.params[_sanitize_keys(k, replaced_dict)].keys()):
+                return_dct[key_u] = v
```

Comment:
this won't work.. the whole point of this function is to "unwrap" the star-notated keys.. so imagine you have Wilson Cowan and you want to address tau parameter. in star notation you can use `model.params[*tau]` so the `k` as the original key would be "*tau" and from this, you wanna get this whole dictionary
```
{
    "WCnode_0.WCmassEXC_0.tau": 2.5,
    "WCnode_0.WCmassEXC_0.noise_0.tau": 2.5,
    "WCnode_0.WCmassINH_1.tau": 2.5,
    "WCnode_0.WCmassINH_1.noise_0.tau": 2.5,
}
```
so the `model.params[_sanitize_keys(k, replaced_dict)].keys()` actually gets all the keys in their full and one by one adds them to `return_dct`
if you could assign more dict keys using a list of keys (which would be your `return_dct[key_r] = v`) then I'd do it, but AFAIK that's not possible

Reply:
btw the try - except is there because the part `model.params[<key not there>]` would throw `AttributeError`

Reply:
The user should _not_ get an `info` if she just decides to `model.params['my_new_par']` though, right?

Reply:
the info would be there for both cases: if the new param contains a star, and even if it doesn't... the AttributeErrors (hence the try-except) are thrown each time if the new parameters are not part of `model.params`

Reply:
So 
`model.params['*tau`] = 5` would set all to `5`,

`model.params['*xyz'] = 5` would lead to a warning/info and set the parameter "*xyz"?

`model.params['xyz'] = 5` would just do what it does

Reply:
but only in MultiModel framework of course.. native neurolib models are not using this function

Reply:
to your question: exactly!

Reply:
I'd suggest let's do an info a la

`logging.info(f"Key `{k}` cannot be resolved")`?

Reply:
Sorry, the last one (`model.params['xyz'] = 5`) should **not** info, because it is a legitimate use case!

Reply:
Only when the user tries to resolve a parameter that does not exist, it should, and just set it, that's what I would expect to happen, what do you think?

Reply:
sure, so something like:
```python
except AttributeError:
            if "*" in k:
                logging.info(f"Key `{k}` cannot be resolved in MultiModel parameters, adding it...")
            # add it "raw"
            return_dct[k] = v
```
good?

Reply:
```
except AttributeError:
    if "*" in k:
        logging.info(f"Key `{k}` cannot be resolved.") # not used for multimodel only ;)
    # add it "raw"
    return_dct[k] = v
```